### ISSUE TITLE
Python 3.10 deprecates distuils ... find alternative for distutils.dir_util.copy_tree()

### ISSUE THREAD

**Issue by sanderjo** [sabnzbd/sabnzbd#1958]
FWIW:

- Python 3.10 has been released today.
- Python 3.10 deprecates distutils: https://www.python.org/dev/peps/pep-0632/
- SABnzbd uses "from distutils.dir_util import copy_tree" at two locations. See https://github.com/sabnzbd/sabnzbd/search?q=distutils

So ... find alternative for distutils.dir_util.copy_tree() ... maybe shutil.copytree() ?

```
$ docker run -it python:3.10-bullseye /bin/bash

root@9b21cabed36d:/# python3.10
Python 3.10.0 (default, Oct  5 2021, 23:39:58) [GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from distutils.dir_util import copy_tree
<stdin>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
>>> 
```

EDIT

Strangely enough, pytest does **not** complain about the distutils / copy_tree in test_postproc.py itself:

```
root@9b21cabed36d:/sabnzbd-3.4.1/tests# pytest test_postproc.py 
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.10.0, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: /sabnzbd-3.4.1/tests
plugins: pyfakefs-4.5.1, httpserver-1.0.1, flaky-3.7.0, tavern-1.16.2, httpbin-1.0.0
collected 1 item                                                                                                                                                           

test_postproc.py .                                                                                                                                                   [100%]

============================================================================ 1 passed in 0.10s =============================================================================

```

but calling test_postproc.py  does complain, as I would expect from python 3.10:

```
root@9b21cabed36d:/sabnzbd-3.4.1/tests# python3.10 test_postproc.py 
/sabnzbd-3.4.1/tests/test_postproc.py:12: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.dir_util import copy_tree
...
```

My assumption: pytest does give warnings about code-under-test, but not about the testing code itself?

**Comment by Cpuroast** [comment]
Also, they switched to 1 universal2 installer for macOS: https://www.python.org/ftp/python/3.10.0/python-3.10.0-macos11.pkg

According to the page it still supports 10.9 and up, would be worth seeing what comes out of the build process with the new version.

**Comment by Safihre** [comment]
@Cpuroast nothing, just the plain old x86 code.
The build chain (setuptools, pip, pyinstaller) is not ready to just make universal2 frozen applications. It works perfect to just run SABnzbd from sources on M1 or regular x86 but to have all depencies available in universal2 format in order to freeze them is not yet easily supported. Currently you need to recompile every single dependency, and we have way too many dependencies to do that. And some have very special compilation requirements (Pyobjc, cryptography, to name 2 big ones).

**Comment by Cpuroast** [comment]
wasn't specifically talking about a universal2 version of sab, there's supposedly perf improvements in 3.10, curious how much it will help.

**Comment by Safihre** [comment]
Aaah check. Well, I doubt it will be much faster.. 
What we really need is this one: https://github.com/python/cpython/pull/25478
But haven't been able to figure it out all weekend 😢

**Comment by sanderjo** [comment]
About python 3.10: "Without quotes, Github Action will detect the Python version as 3.1, not 3.10 lol. It's due to the YAML property"
https://github.com/psf/pyperf/commit/a5da5ffccd6ede2a08b6acc8447b4aef2c58ebcb

**Comment by jcfp** [comment]
@sanderjo did you encounter problems with shutil?

**Comment by sanderjo** [comment]
> @sanderjo did you encounter problems with shutil?

No problems/warnings while running SABnzbd with python3.10, to my surprise.
But ... I did not run the sab self-test suite, which contains the distutils.dir_util.copy_tree().

And in the meantime I can't remember how I got that setup with a python3.10 ... In a docker, with python 3.10 from deadsnakes PPA, with a python venv, with modules for python3.10 installed via pip ... ? Not an easy setup.
